sdk/eidolon_ai_sdk/agent/vectara_agent.py

```diff
@@ -0,0 +1,92 @@
+import copy
+import json
+import os
+from typing import Annotated
+from urllib.parse import urljoin
+
+from fastapi import Body
+from httpx import AsyncClient
+from httpx_sse import EventSource
+from pydantic import BaseModel, Field
+
+from eidolon_ai_client.events import StringOutputEvent, StartStreamContextEvent, ObjectOutputEvent, \
+    EndStreamContextEvent, AgentStateEvent
+from eidolon_ai_sdk.agent.agent import register_action
+from eidolon_ai_sdk.system.processes import MongoDoc
+from eidolon_ai_sdk.system.reference_model import Specable
+
+
+os.environ.setdefault("VECTARA_API_KEY", "test")
+
+
+class VectaraAgentSpec(BaseModel):
+    corpus_key: str
+    description: str = "Search documents related to {{ corpus_key }}"
+    vectara_url: str = "https://api.vectara.io/"
+    body_overrides: dict = Field({}, description="Arguments to use when creating / continuing a chat. See https://docs.vectara.com/docs/rest-api/create-chat for more information.")
```

Comment:
Agent Configuration lives here. Override allow the creation / turn interface to be arbitrarily customized since we don't know how people will want to use This. contract is the same for both endpoints,